app/components/Views/confirmations/hooks/useConfirmationRedesignEnabled.ts

```diff
@@ -67,6 +68,10 @@ function isRedesignedTransaction({
     return confirmationRedesignFlags?.staking_transactions;
   }
 
+  if (transactionMetadata?.type === TransactionType.contractInteraction) {
+    return confirmationRedesignFlags?.contract_interaction;
```

Comment:
```suggestion
    return (
       process.env.TEMP_INCOMPLETE_REDESIGNED_CONFIRMATION &&
       confirmationRedesignFlags?.contract_interaction
    );
```

Based on the [latest discussions](https://consensys.slack.com/archives/C070G0RCX0T/p1743522944564149), we need to add a condition to handle incomplete redesigned confirmations. Relying solely on feature flags for an incomplete confirmation in a release makes the flag unusable in future releases, as enabling it would also activate it for past versions.

This temporary environment variable will be removed once we confirm that the contractInteraction redesigned confirmations are complete.

Reply:
As discussed in slack conversation, we can use one single env variable per confirmation in the following way:

whenever a confirmation is finalized and ready to be tested in the RC we create a PR that removes this code:
```
const isContractInteractionEnabled = getFeatureFlagValue(
      process.env.FEATURE_FLAG_REDESIGNED_CONTRACT_INTERACTION,
      confirmationRedesignFlags.contract_interaction,
```
and changes it to this one without any environment variable for that confirmation:
```
const isContractInteractionEnabled = confirmationRedesignFlags.contract_interaction;
```
We also enable the remote feature flag on dev and rc so that:
- For this new version Barbara can test in the RC, and we can see it locally as well
- For older versions already deployed to our users, no one can see the incomplete confirmation because the environment variable defaults to false